### PR TITLE
[Feat][Training] Rename weight conversion function and update gradient checkpoint in scripts

### PR DESCRIPTION
**Pull request by BrianChen1129**

### CODE REVIEW

File: fastvideo/v1/training/training_utils.py [hunk 1]
```diff
@@ -487,8 +487,8 @@ def _has_foreach_support(tensors: List[torch.Tensor],
         t is None or type(t) in [torch.Tensor] for t in tensors)
 
 
-def convert_training_to_diffusers_format(state_dict: Dict[str, Any],
-                                         transformer) -> Dict[str, Any]:
+def convert_custom_format_to_diffusers_format(state_dict: Dict[str, Any],
+                                              transformer) -> Dict[str, Any]:
     """
     Convert training format state dict to diffusers format using reverse_param_names_mapping.
```
Comment (Edenzzzz) [inline fastvideo/v1/training/training_utils.py]: Please also update doc string

**Comment by BrianChen1129** [comment]
passed all the tests and update comment
